extensions/amp-carousel/0.1/base-carousel.js

```diff
@@ -25,17 +25,66 @@ export class BaseCarousel extends AMP.BaseElement {
     /** @private {!Element} */
     this.nextButton_;
 
-    this.buildCarousel();
-    this.buildButtons();
-    this.setupGestures();
-    this.setControlsState();
+    /** @private {?number} */
+    this.autoplayTimeoutId_ = null;
```

Comment:
How would we extend autoplay without slides? I think this is a wrong place for this functionality. I'd be fine if we have a separate base class for slides, of course, if that makes sense. But scroller doesn't have any of these features in common.